src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,146 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (size == 0) {
+            first = last = newNode;
+        } else {
+            newNode = new Node<>(last, value, null);
+            last.next = newNode;
+            last = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Objects.checkIndex(index, size + 1);
+        Node<T> newNode = new Node<>(null, value, null);
+        Node<T> node = first;
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (index == 0) {
+            newNode = new Node<>(null, value, first);
+            newNode.next = first;
+            first = newNode;
+        } else {
+            for (int i = 0; i < index - 1; i++) {
+                node = node.next;
+            }
+            newNode.next = node.next;
+            newNode.prev = node;
+            node.next.prev = newNode;
+            node.next = newNode;
+        }
+        size++;
+
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        Objects.checkIndex(index, size);
+        Node<T> node = first;
```

Comment:
You can use the getNodeByIndex method here too. Also put an index check inside the getNodeByIndex method